src/window/gui/Gui.cpp

```diff
@@ -328,6 +328,16 @@ void Gui::DrawMenu() {
     ImGuiWMNewFrame();
     ImGui::NewFrame();
 
+    // Get the difference between the last frame and the current Frame
+    mDpiScaleDiff = GImGui->CurrentDpiScale / mLastDpiScale;
+    // Store the current scale for next frame
+    mLastDpiScale = GImGui->CurrentDpiScale;
+
+    // Apply the scale difference to ImGui
+    ImGui::GetStyle().ScaleAllSizes(mDpiScaleDiff);
+    ImFont* font = ImGui::GetFont();
+    font->Scale *= mDpiScaleDiff;
```

Comment:
doing this every frame feels a bit like overkill to me. it seems on windows we could hook into `WM_DPICHANGED`, and i assume SDL has something similar (i'm not 100% sure what's going on in https://github.com/Kenix3/libultraship/pull/365/ but it seems like it , maybe @Spodi can explain a bit?)

Reply:
Yes that is a valid point. But i think this is already a good solution. If we would put that into an window event, we also need an extra initialization for the scale. And we would need to call Gui functions from Fast3D files since the message loops are located there. I think that would be to over complicated for this.
To prevent unnecessary changes to the scale we just could change the scale if the difference is != 1,0.
```suggestion
    // Get the difference between the last frame and the current Frame
    mDpiScaleDiff = GImGui->CurrentDpiScale / mLastDpiScale;
    
    // Apply the scale difference to ImGui
    if (mDpiScaleDiff != 1.f) {
        // Store the current scale for next frame
        mLastDpiScale = GImGui->CurrentDpiScale;
        ImGui::GetStyle().ScaleAllSizes(mDpiScaleDiff);
        ImFont* font = ImGui::GetFont();
        font->Scale *= mDpiScaleDiff;
    }
```

Reply:
I've no clue where ImGUI gets the DPI-scale.
On dxgi you can do it inside of `WM_DPICHANGED`.
Not sure if there is an equivalent in SDL. I don't think SDL has its own event for this, but you can get the physical dpi.
 For Windows + SDL you could add `WM_DPICHANGED` to the hook I already did for the Dpi stuff in there to prevent some weirdness.

Reply:
Oh actually SDL already handles `WM_DPICHANGED`, so you can't just overwrite stuff with my hook. You would probably need to pass the event to SDLs handler too.